R/read_csv.R

```diff
@@ -321,6 +321,9 @@ read_cmdstan_csv <- function(files,
   metadata$inv_metric <- NULL
   metadata$lines_to_skip <- NULL
   metadata$model_params <- repair_variable_names(metadata$model_params)
+  model_param_dims <- variable_dims(metadata$model_params)
+  metadata$model_params_dims <- model_param_dims
+  metadata$model_params_named <- names(model_param_dims)
```

Comment:
For the names what about using `metadata$variable_dims` and `metadata$variable_names`?

Reply:
And then matching fit$variable_dims()? That works for me.

Reply:
I'm ok with that, but I wonder if we need both `fit$metadata()` _and_ methods for accessing elements of `fit$metadata()`. Could we just tell the user to use `fit$metadata()$variable_dims` if they want it? That way we don't need to provide methods for accessing elements of the return object of other methods.  What do you think? I don't feel too strongly about it but worth considering before we add new methods.

Reply:
Yes. I am fine with that. Good call.